src/api/File.js

```diff
@@ -77,6 +78,26 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * API for getting a thumbnail URL for a BoxItem
+     *
+     * @param {BoxItem} item - BoxItem to get the thumbnail URL for
+     * @return {Promise<?string>} - the url for the item's thumbnail, or null
+     */
+    async getThumbnailUrl(item: BoxItem): Promise<?string> {
+        const entry = getProp(item, 'representations.entries[0]');
+        const extension = getProp(entry, 'representation');
+        const template = getProp(entry, 'content.url_template');
+        const token = await TokenService.getReadToken(getTypedFileId(item.id), this.options.token);
+
+        if (getProp(entry, 'status.state') !== 'success' || !extension || !template || !token) {
+            return null;
+        }
+
+        const thumbnailBaseUrl = template.replace('{+asset_path}', extension === 'jpg' ? '' : '1.png');
+        return `${thumbnailBaseUrl}?access_token=${token}`;
```

Comment:
Consider parsing the original tempalte url using `queryString.parseUrl` to retain any existing query parameters and also `queryString.stringify` to produce the final string. Take a look at `getDownloadUrl` for an example.